### PR TITLE
feat: setup e2e env

### PR DESCRIPTION
**Pull request by daniel-hutao**
Signed-off-by: Daniel Hu <tao.hu@merico.dev>

# Summary

feat: setup e2e env

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,10 +20,14 @@ dtm
 .glide/
 
 # YAML for testings
-*.yaml
+config*.yaml
```
Comment (daniel-hutao) [inline .gitignore:23]: Let git catch the `kind.yaml` file

**Comment by daniel-hutao** [comment]
image

image

image

File: hack/e2e/e2e-up.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# kubectl & kind
+function fetch_tools() {
+  echo "All tools save in ${WORK_DIR} directory"
+  if [ ! -f ${WORK_DIR} ]; then
+    mkdir -p ${WORK_DIR}
+  fi
+  cd ${WORK_DIR}
+
+  if [ ! -f kind ]; then
+    echo "kind not exists, download it now"
```
Comment (IronCore864) [inline hack/e2e/e2e-up.sh:61]: kind not exists -> kind doesn't exist

Comment (IronCore864) [reply hack/e2e/e2e-up.sh:61]: same for kubectl message

**Comment by IronCore864** [comment]
Please add a simple readme under the "hack" folder. Example:

https://github.com/kubernetes/kubernetes/blob/master/hack/README.md

Please also mention the existence of the "hack" folder in the project layout document: https://github.com/merico-dev/stream/blob/main/docs/project_layout.md.

File: hack/e2e/e2e-up.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    kind_uri="https://kind.sigs.k8s.io/dl/v${KIND_VERSION}/kind-${HOST_OS}-${HOST_ARCH}"
+    echo "kind uri: ${kind_uri}"
+    curl -Lo ./kind "${kind_uri}"
+  fi
+  echo "kind already downloaded"
+  chmod +x ./kind
+
+  if [ ! -f kubectl ]; then
+    echo "kubectl not exists, download it now"
```
Comment (lfbdev) [inline hack/e2e/e2e-up.sh:70]: the same to “kind not exists”

File: hack/e2e/e2e-up.sh [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    echo "Support Darwin/Linux OS only."
+    exit 1
+  fi
+
+  if [ "$(arch)" == "amd64" ];then
+    HOST_ARCH="arm64"
+  elif [ "$(arch)" == "arm64" ];then
+    HOST_ARCH="amd64"
+  else
+    echo "Support amd64/arm64 cpu arch only."
```
Comment (lfbdev) [inline hack/e2e/e2e-up.sh:36]: "cpu" lowercase or uppercase? like the following "and CPU arch:"

**Review by lfbdev — APPROVED** [review]
/lgtm

File: hack/e2e/e2e-up.sh [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+# kubectl & kind
+function fetch_tools() {
+  echo "All tools save in ${WORK_DIR} directory"
+  if [ ! -f ${WORK_DIR} ]; then
+    mkdir -p ${WORK_DIR}
+  fi
+  cd ${WORK_DIR}
+
+  if [ ! -f kind ]; then
+    echo "kind doesn't exists, download it now"
```
Comment (IronCore864) [inline hack/e2e/e2e-up.sh:61]: kind doesn't exists -> kind doesn't exist; similar for kubectl

File: hack/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# DevStream hack GuideLines
```
Comment (IronCore864) [inline hack/README.md:1]: Very good

**Review by IronCore864 — APPROVED** [review]

**Pull request merged**